### ISSUE TITLE
Question re: Implementation of `CoopGame._gamma` method

### ISSUE THREAD

**Issue by rolandmaio** [lqgentner/PyGameTheory#1]
Hi,

I'm implementing the Shapley value in one of my own projects. ChatGPT mentioned your project when I asked about Python libraries for computing the Shapley value. I looked through your code for ideas and pointers. It's been super helpful. Your code is very nice and clean!

But there is one thing I am confused about. If I understand correctly, the `CoopGame._gamma` method computes the weight of the subcoalition-specific summands in the Shapley value. The current expression for the weight is `factorial(n-s) * factorial(s-1) / factorial(n)`. And again, if I understand correctly, `n` is a numpy array whose every entry is the total number of players, i.e. the integer n, whereas `s` is a numpy array where the i-th entry is associated to a fixed subcoalition and is that subcoalition's size. If so, then based on the formula for the Shapley value, shouldn't the numerator of the weight be `factorial(n - s - 1)*factorial(s)`?

If I'm wrong, I'm curious to understand why. And otherwise, I figured it would be worth bringing to your attention.

Thanks!
Roland

**Comment by lqgentner** [comment]
Hi Roland,

Thanks for your message! I am humbled that ChatGPT recommended my repository! 😄 Are you sure it was really this one and not another repository with the same name? I thought ChatGPT had a knowledge cutoff sometime in 2022. I'd like to know what you asked it because I couldn't get it to recommend my repository. :D

First of all, I have no background in mathematics or game theory. I am just an aerospace engineering student and made this little library for a class during my Erasmus semester, so take everything with a grain of salt 😉.

You are right that `n` is the number of players (a scalar) and `s` is a list of sub coalition sizes.

Here are the lecture slides I used for reference. I remember that it took me a while to understand that there are several equivalent formulas to compute the Shapely values, and some are easier to compute by hand and some are easier to implement in code. So maybe your version will end up with the same results? Below I have annotated where I got my gamma formula from:

![image](https://github.com/lqgentner/PyGameTheory/assets/61919484/572570fa-7a27-479d-8176-27d692d2b49e)
![image](https://github.com/lqgentner/PyGameTheory/assets/61919484/1a3ff061-a9ef-48f7-84ac-1f6147151818)
![image](https://github.com/lqgentner/PyGameTheory/assets/61919484/e07e3eba-48be-4a70-845b-ad0754141f84)

For a three-player game, n = 3 and s = [1, 2, 2, 3]. Both vars are converted to numpy arrays, which allow element-wise operations. γ is then calculated to gamma = [0.333, 0.167, 0.167, 0.333].

I hope this helps.

Good luck with your further endeavors in game theory,
Luis
